.github/workflows/nightly.yml

```diff
@@ -94,3 +94,51 @@ jobs:
       s3_bucket: ${{ secrets.IT_CASE_S3_BUCKET }}
       s3_access_key: ${{ secrets.IT_CASE_S3_ACCESS_KEY }}
       s3_secret_key: ${{ secrets.IT_CASE_S3_SECRET_KEY }}
+
+  build_python_wheels:
+    name: "Build Python Wheels on ${{ matrix.os }}"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-latest
+            os_name: linux
+            python-version: 3.9
+          - os: macos-latest
+            os_name: macos
+            python-version: 3.9
```

Comment:
Looks like the python version also got changed in [FLINK-34582](https://issues.apache.org/jira/browse/FLINK-34582) 🤔 We should check why we don't upgrade the Python version for Linux.

Reply:
Yes, indeed, the linux version of build scripts uses versions until 3.10 but not 3.11. I have asked on the ticket.

https://github.com/HuangXingBo/flink/blob/master/flink-python/dev/build-wheels.sh#L19-L26

Reply:
> You're checking [Xingbo Huang](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hxb)'s fork where the master branch doesn't seem to be up-to-date. [apache/flink:flink-python/dev/build-wheels.sh](https://github.com/apache/flink/blob/master/flink-python/dev/build-wheels.sh#L19-L26) does, indeed, have 3.11 added to the python version list.

Ok, I [responded in the Jira issue](https://issues.apache.org/jira/browse/FLINK-34582?focusedCommentId=17849185&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17849185).

Looks like the Python version doesn't make a diffrence here for Linux. We might want to add this as a comment and refer to the build script stating that the version is specified there.

In the mean time: Can you find out where the Python versions are configured for the MacOS builds? It appears that we also have the very same list of Python versions used for building macos wheels (see [this Azure pipeline run|https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=59794&view=logs&j=f73b5736-8355-5390-ec71-4dfdec0ce6c5&t=90f7230e-bf5a-531b-8566-ad48d3e03bbb&l=49] as an example). I only found [apache/flink:flink-python/pyproject.toml:30|https://github.com/apache/flink/blob/942d234444636b13df7a118274d307943f5524cd/flink-python/pyproject.toml#L30]. That might be the right configuration. But I don't understand why the Python version needed to be updated in the workflow yaml to 3.11 in FLINK-34582 in that case. We might need a comment there to clarify things for macos as well.

Reply:
Yes on macos the only pyproject.toml is used by cibuildwheel. Tools seems to be well documented, maybe we could also use it for linux builds.

It uses the `cp38-*`, etc for Python versions, https://cibuildwheel.pypa.io/en/stable/options/#build-skip. For Linux we are running bash scripts for building wheel, but we could let the cibuildwheel action do for both platforms.

We can ask @HuangXingBo for final review